src/hooks/swap/useSwapInfo.tsx

```diff
@@ -97,11 +101,16 @@ function useComputeSwapInfo(routerUrl?: string): SwapInfo {
   const slippage = useSlippage(trade)
   const impact = usePriceImpact(trade.trade)
 
+  const permit2 = usePermit2()
   const maximumAmountIn = useMemo(() => {
     const maximumAmountIn = trade.trade?.maximumAmountIn(slippage.allowed)
     return maximumAmountIn?.currency.isToken ? (maximumAmountIn as CurrencyAmount<Token>) : undefined
   }, [slippage.allowed, trade.trade])
-  const approval = useSwapApproval(maximumAmountIn)
+  const approval = useSwapApproval(permit2 ? undefined : maximumAmountIn)
```

Comment:
a bit confused what's happening here

Reply:
passing undefined if it is not the code path we want, to prevent nested hooks from executing and fetching on-chain or off-chain data